src/Simulation/Common/SimulatorBase.cs

```diff
@@ -429,8 +425,7 @@ public GetQubitsAvailableToBorrow(SimulatorBase m) : base(m)
                 sim = m;
             }
 
-            public override Func<QVoid, long> __Body__ => (arg) => sim.QubitManager.GetParentQubitsAvailableToBorrowCount() +
-                                                               sim.QubitManager.GetFreeQubitsCount();
+            public override Func<QVoid, long> __Body__ => (arg) => sim.QubitManager.GetQubitsAvailableToBorrowCount(1);
```

Comment:
If passing `1` means get the value for the current stack frame, perhaps it's worth having that as the default value for the parameter?

Reply:
Technically, passing 1 gets the value for the parent stack frame rather than the current one. The reason this needs to be 1 here however is that in Q# you would have a call to this operation and then *within* the called operation, the value is evaluated, meaning to get the accurate result you would need to know not the qubits in the current stack frame (which is inside GetQubitsAvailableToBorrow), but rather the qubits in the parent (which is where GetQubitsAvailableToBorrow is called).